File: BubbleSortOptimized.py
#!/env/bin/python

import random

n=100

def BubbleSort(A):
    m=0
    for i in range(0,n-1):
        for j in range(1,n-i):
            if A[j]<A[j-1]:
                A[j],A[j-1]=A[j-1],A[j]
                m=1
        if m==0: break
        m=0
    return A
    

if __name__=='__main__':
    A=[0 for i in range(n)]

    for i in range(n):
        A[i]=random.randint(0,n-1)

    print("Los numeros a ordenar son\n")
    print(A)
        
    A=BubbleSort(A)
    print("\n\nLos numeros ordenados son\n")
    print(A)
    

File: BubbleSortOptimized.cpp
#include<stdio.h>
#include<time.h>
#include<stdlib.h>

#define N 10000

void Buble(int A[N]);

int main(){
	int A[N];

	//printf("Lista de numeros a ordenar\n\n");

	for(int i=0; i<N;i++){
		A[i]= rand()%N;
		//printf("%i ", A[i]);
	}

	Buble(A);
	printf("\n\n");
	//system("pause");
	return 0;

}

void Buble(int A[N]){
	int aux;
	int m=0;

   for(int i=0;i<N-1;i++){
		for(int j=1;j<N-i;j++){
			if(A[j-1]>A[j]){
				aux=A[j];
				A[j]=A[j-1];
				A[j-1]=aux;
				m=1;
			}
		}
		if(m==0) break;
		m=0;
	}

   printf("\n\nLos numeros ordenados son: \n\n");

	for(int i=0; i<N; i++){
		printf("%i ", A[i]);
	}

}
